**<h1 style="color:red;">📊 Načítanie Datasetu</h1>**

In [ ]:
import pandas as pd
import numpy as np
df=pd.read_csv(r"/content/instagram_usage_lifestyle.csv")
df.head()

user_id   app_name  age      gender         country urban_rural  \
0        1  Instagram   51      Female           India       Rural   
1        2  Instagram   64      Female  United Kingdom       Urban   
2        3  Instagram   41      Female          Canada       Urban   
3        4  Instagram   27  Non-binary     South Korea       Urban   
4        5  Instagram   55        Male           India       Urban   

   income_level   employment_status education_level relationship_status  ...  \
0          High             Retired      Bachelor’s              Single  ...   
1        Middle  Full-time employed           Other            Divorced  ...   
2        Middle             Student      Bachelor’s   In a relationship  ...   
3        Middle          Unemployed        Master’s   In a relationship  ...   
4  Upper-middle  Full-time employed      Bachelor’s              Single  ...   

  last_login_date  average_session_length_minutes  content_type_preference  \
0      2025-11-02                             5.0                    Mixed   
1      2025-03-22                            14.8                   Photos   
2      2025-08-10                             5.0                    Mixed   
3      2025-03-31                            25.9                  Stories   
4      2025-03-19                            13.1                   Videos   

  preferred_content_theme privacy_setting_level two_factor_auth_enabled  \
0                    Tech               Private                     Yes   
1                 Fashion                Public                      No   
2                   Other                Public                     Yes   
3                    Tech               Private                      No   
4                    Food                Public                     Yes   

   biometric_login_used  linked_accounts_count  subscription_status  \
0                    No                    0.0                 Free   
1                    No                    3.0                 Free   
2                   Yes                    1.0                 Free   
3                    No                    1.0                 Free   
4                    No                    0.0                 Free   

   user_engagement_score  
0                   7.83  
1                   1.43  
2                   9.67  
3                   0.94  
4                   1.03  

[5 rows x 58 columns]

# **Základné údaje o Datasete**📊

In [ ]:
df.shape # koľko ma dataset riadkov
df.columns # stručny prehľad stlpcov v datasete
df.isna().value_counts().sum()  # zistíme koľko má dataset prázdnych hodnôt (V prípade potreby vieme doplniť alebo odstraniť - df.fillna(0), df.dropna())
df.describe()
df.info()  # špecifické informácie ako dátove typy, (float, string, int, bool)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18580 entries, 0 to 18579
Data columns (total 58 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   user_id                         18580 non-null  int64  
 1   app_name                        18580 non-null  object 
 2   age                             18580 non-null  int64  
 3   gender                          18580 non-null  object 
 4   country                         18580 non-null  object 
 5   urban_rural                     18580 non-null  object 
 6   income_level                    18580 non-null  object 
 7   employment_status               18580 non-null  object 
 8   education_level                 18580 non-null  object 
 9   relationship_status             18580 non-null  object 
 10  has_children                    18580 non-null  object 
 11  exercise_hours_per_week         18580 non-null  float64
 12  sleep_hours_per_night           

# **Úprava Datasetu**🧪

In [ ]:
# V úprave si premenujeme celý Dataset do slovenského jazyka (pre lepšiu orientaciu)
# Taktiež si premenujeme aj vnútorne bunky každého stplca


In [ ]:
# Prepísanie stlpcov z ANJ do SJ
df = df.rename(columns={
    "user_id": "id_pouzivatela",
    "app_name": "nazov_aplikacie",
    "age": "vek",
    "gender": "pohlavie",
    "country": "krajina",
    "urban_rural": "mesto_vidiek",
    "income_level": "uroven_prijmu",
    "employment_status": "pracovny_status",
    "education_level": "vzdelanie",
    "relationship_status": "partnersky_status",
    "has_children": "ma_deti",
    "exercise_hours_per_week": "hodiny_cvicenia_za_tyden",
    "sleep_hours_per_night": "hodiny_spanku_za_noc",
    "diet_quality": "kvalita_stravy",
    "smoking": "fajcenie",
    "alcohol_frequency": "frekvencia_alkoholu",
    "perceived_stress_score": "vnimany_stres",
    "self_reported_happiness": "subjektivne_stastie",
    "body_mass_index": "BMI",
    "blood_pressure_systolic": "krvny_tlak_systolicky",
    "blood_pressure_diastolic": "krvny_tlak_diastolicky",
    "daily_steps_count": "kroky_za_den",
    "weekly_work_hours": "pracovne_hodiny_za_tyden",
    "hobbies_count": "pocet_zalub",
    "social_events_per_month": "spolocenske_akcie_za_mesiac",
    "books_read_per_year": "knihy_precitane_za_rok",
    "volunteer_hours_per_month": "dobrovolnictvo_hodiny_za_mesiac",
    "travel_frequency_per_year": "cestovanie_za_rok",
    "daily_active_minutes_instagram": "aktivne_minuty_na_instagrame_za_den",
    "sessions_per_day": "pocet_relacii_za_den",
    "posts_created_per_week": "prispevky_za_tyden",
    "reels_watched_per_day": "reels_pozrete_za_den",
    "stories_viewed_per_day": "stories_pozrete_za_den",
    "likes_given_per_day": "lajky_za_den",
    "comments_written_per_day": "komentare_za_den",
    "dms_sent_per_week": "spravy_odoslane_za_tyden",
    "dms_received_per_week": "spravy_prijate_za_tyden",
    "ads_viewed_per_day": "reklamy_zobrazene_za_den",
    "ads_clicked_per_day": "reklamy_kliknute_za_den",
    "time_on_feed_per_day": "cas_na_feede_za_den",
    "time_on_explore_per_day": "cas_na_explore_za_den",
    "time_on_messages_per_day": "cas_na_spravach_za_den",
    "time_on_reels_per_day": "cas_na_reels_za_den",
    "followers_count": "pocet_followers",
    "following_count": "pocet_following",
    "uses_premium_features": "pouziva_premium",
    "notification_response_rate": "reakcia_na_notifikacie",
    "account_creation_year": "rok_vytvorenia_uctu",
    "last_login_date": "posledne_prihlasenie",
    "average_session_length_minutes": "priemerna_dlzka_relacie_min",
    "content_type_preference": "preferovany_typ_obsahu",
    "preferred_content_theme": "preferovana_tema_obsahu",
    "privacy_setting_level": "uroven_sukromia",
    "two_factor_auth_enabled": "dvojfaktorove_overenie",
    "biometric_login_used": "biometricke_prihlasenie",
    "linked_accounts_count": "pocet_prepojenych_uctov",
    "subscription_status": "stav_predplatneho",
    "user_engagement_score": "skore_angazovanosti"
})

In [ ]:
df["pohlavie"].unique()  # ukáže nám unikátne názvy v kategorií pohlavie - aby sme mohli zmeniť názvy do SJ

array(['Female', 'Non-binary', 'Male', 'Prefer not to say'], dtype=object)

In [ ]:
#zmena vnútornej bunky z anglických názvov na slovenské
df["pohlavie"].replace({
    "Male" : "Muž",
    "Female":"Žena",
    "Non-binary":"Nebinárny",
    "Prefer not to say":"Preferuje_nepovedať"
}, inplace=True)

/tmp/ipython-input-3957190003.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["pohlavie"].replace({


In [ ]:
# Úprava vnútornej bunky do Slovenčiny
df["krajina"].replace({
    "Brazil":"Brazília",
    "Canada":"Kanada",
    "Germany":"Nemecko",
    "Japan":"Japonsko",
    "Other":"Iné",
    "South Korea":"Južná Koreja",
    "Spain":"Španielsko",
    "United States":"Spojené státy americké",
    "United Kingdom":"Spojené královstvo"

}, inplace=True)

/tmp/ipython-input-1095880424.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["krajina"].replace({


In [ ]:
df["mesto_vidiek"].replace({
    "Urban":"Mesot",
    "Rural":"Vidiek",
    "Suburban":"Predmestie"
}, inplace=True)


/tmp/ipython-input-1564574229.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["mesto_vidiek"].replace({


In [ ]:
df["uroven_prijmu"].replace({
    "Low":"Nízky",
    "Medium":"Stredný",
    "High":"Vysoký"
}, inplace=True)

/tmp/ipython-input-2381710476.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["uroven_prijmu"].replace({


In [ ]:
df["pracovny_status"].replace({
    "Employed":"Zamestnanec",
    "Self-employed":"Živnostník",
    "Unemployed":"Nezamestnanec",
    "Student":"Student",
    "Retired":" Dôchodca",
    "Freelancer":"Freelancer",
    "Full-time employed":"Full-time",
    "Part-time":"Part-time"
}, inplace=True)


/tmp/ipython-input-867943899.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["pracovny_status"].replace({


In [ ]:
df["vzdelanie"].replace({
    "Bachelor’s":"Bakalár",
    "Other":"Iné",
    "Master’s":"Magister",
    "High school":"Stredná škola",
    "Some college":"Nezaradená škola",
    "PhD": "Doktorát"
 },inplace=True)

/tmp/ipython-input-4174906071.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["vzdelanie"].replace({


In [ ]:
df["partnersky_status"].replace({
    "Single":"Slobodný",
    "Divorced":"Rozvedený",
    "In a relationship":"Zadaný",
    "Married":"Ženatý",
    "Widowed":"Vdovec"
}, inplace=True)



/tmp/ipython-input-3949590101.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["partnersky_status"].replace({


In [ ]:
df["preferovana_tema_obsahu"].replace({
    "Tech":"Technologie",
    "Fasion":"Móda",
    "Other":"Iné",
    "Food":"Jedlo",
    "Travel":"Cestovanie",
    "Art":"Umenie",
    "Music":"Muzika",
    "Fitness":"Fitnes",
}, inplace=True)


/tmp/ipython-input-3611187554.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["preferovana_tema_obsahu"].replace({


In [ ]:
df["uroven_sukromia"].replace({
    "Private":"Súkromné",
    "Public":"Verejné",
    "Friends only":"Len priatelia",

},inplace=True)


/tmp/ipython-input-3487933343.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["uroven_sukromia"].replace({


In [ ]:
df["dvojfaktorove_overenie"].replace({
    "Yes":"ANO",
    "No":"NIE"
    },inplace=True)


/tmp/ipython-input-860998636.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["dvojfaktorove_overenie"].replace({


In [ ]:
df["biometricke_prihlasenie"].replace({
    "No","NIE",
    "Yes","ANO"
},inplace=True)

/tmp/ipython-input-347115792.py:1: FutureWarning: Series.replace without 'value' and with non-dict-like 'to_replace' is deprecated and will raise in a future version. Explicitly specify the new values instead.
  df["biometricke_prihlasenie"].replace({
/tmp/ipython-input-347115792.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["biometricke_prihlasenie"].replace({


In [ ]:
df["stav_predplatneho"].replace({
    "Free":"Zadarmo",
    "Premium":"Premium",

},inplace=True)



/tmp/ipython-input-818200589.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df["stav_predplatneho"].replace({


In [ ]:
df.tail()

id_pouzivatela nazov_aplikacie  vek pohlavie                 krajina  \
18575           18576       Instagram   37      Muž  Spojené státy americké   
18576           18577       Instagram   44      Muž                Brazília   
18577           18578       Instagram   20     Žena               Australia   
18578           18579       Instagram   17      Muž                     Iné   
18579           18580       Instagram   42      Muž                   India   

      mesto_vidiek uroven_prijmu pracovny_status         vzdelanie  \
18575   Predmestie  Lower-middle         Student               Iné   
18576        Mesot        Middle       Full-time           Bakalár   
18577        Mesot        Middle       Full-time           Bakalár   
18578   Predmestie           Low       Full-time     Stredná škola   
18579        Mesot          High         Student  Nezaradená škola   

      partnersky_status  ... posledne_prihlasenie  \
18575          Slobodný  ...           2025-02-01   
18576            Ženatý  ...           2025-12-17   
18577            Ženatý  ...           2025-11-17   
18578         Rozvedený  ...           2025-07-26   
18579          Slobodný  ...                  NaN   

       priemerna_dlzka_relacie_min  preferovany_typ_obsahu  \
18575                         10.5                   Reels   
18576                         30.9                   Reels   
18577                         10.5                  Photos   
18578                         27.0                 Stories   
18579                          NaN                     NaN   

      preferovana_tema_obsahu uroven_sukromia dvojfaktorove_overenie  \
18575             Technologie         Verejné                    NIE   
18576                   Jedlo   Len priatelia                    NIE   
18577             Technologie   Len priatelia                    ANO   
18578                  Umenie   Len priatelia                    ANO   
18579                     NaN             NaN                    NaN   

       biometricke_prihlasenie  pocet_prepojenych_uctov  stav_predplatneho  \
18575                       No                      3.0            Premium   
18576                       No                      0.0           Business   
18577                       No                      2.0            Zadarmo   
18578                      Yes                      5.0            Zadarmo   
18579                      NaN                      NaN                NaN   

       skore_angazovanosti  
18575                 0.94  
18576                 1.01  
18577                 1.11  
18578                 1.11  
18579                  NaN  

[5 rows x 58 columns]

# **Vytvorenie a uloženie čistého Datasetu**📊

In [ ]:
# Do numerických stĺpcov vložíme pri hodnote NAN - hodnotu 0 - pomocou funkcie fillna()
# Do stringových stĺpcov necháme hodnotu NAN
# Na záver uložíme jeden čisty dataset - pripravený na ďalšiu prácu

In [ ]:
df[df.select_dtypes(include=["int64", "float64"]).columns] = \
    df[df.select_dtypes(include=["int64", "float64"]).columns].fillna(0)

In [ ]:
df.to_csv("clean_dataset.csv",index=False)

# Dataset (DF) - je uložený a pripravený na následné úlohy

# **Analýza správania používateľov**📊
V tejto časti sa budeme primárne zameravať na odpoveď na následujúce otázky:

1. Koľko je mužov, žien a ostatných používateľov?

2. Aký je priemerný vek mužov a žien?

3. V ktorých krajinách ľudia trávia na Instagrame najviac času za deň?

4. Trávia ľudia s vyšším príjmom viac alebo menej času na Instagrame?

5. Sú ľudia z miest na Instagrame aktívnejší ako ľudia z vidieka?

6. Ktoré pracovné skupiny (študenti, zamestnaní, nezamestnaní) sú na Instagrame najaktívnejšie?






In [ ]:
# Importovanie vyčisteného datasetu
import pandas as pd
import numpy as np
df1=pd.read_csv(r"/content/clean_dataset.csv")
df1.head()

id_pouzivatela nazov_aplikacie  vek   pohlavie             krajina  \
0               1       Instagram   51       Žena               India   
1               2       Instagram   64       Žena  Spojené královstvo   
2               3       Instagram   41       Žena              Kanada   
3               4       Instagram   27  Nebinárny        Južná Koreja   
4               5       Instagram   55        Muž               India   

  mesto_vidiek uroven_prijmu pracovny_status vzdelanie partnersky_status  ...  \
0       Vidiek        Vysoký        Dôchodca   Bakalár          Slobodný  ...   
1        Mesot        Middle       Full-time       Iné         Rozvedený  ...   
2        Mesot        Middle         Student   Bakalár            Zadaný  ...   
3        Mesot        Middle   Nezamestnanec  Magister            Zadaný  ...   
4        Mesot  Upper-middle       Full-time   Bakalár          Slobodný  ...   

  posledne_prihlasenie  priemerna_dlzka_relacie_min  preferovany_typ_obsahu  \
0           2025-11-02                          5.0                   Mixed   
1           2025-03-22                         14.8                  Photos   
2           2025-08-10                          5.0                   Mixed   
3           2025-03-31                         25.9                 Stories   
4           2025-03-19                         13.1                  Videos   

  preferovana_tema_obsahu uroven_sukromia dvojfaktorove_overenie  \
0             Technologie        Súkromné                    ANO   
1                 Fashion         Verejné                    NIE   
2                     Iné         Verejné                    ANO   
3             Technologie        Súkromné                    NIE   
4                   Jedlo         Verejné                    ANO   

   biometricke_prihlasenie  pocet_prepojenych_uctov  stav_predplatneho  \
0                       No                      0.0            Zadarmo   
1                       No                      3.0            Zadarmo   
2                      Yes                      1.0            Zadarmo   
3                       No                      1.0            Zadarmo   
4                       No                      0.0            Zadarmo   

   skore_angazovanosti  
0                 7.83  
1                 1.43  
2                 9.67  
3                 0.94  
4                 1.03  

[5 rows x 58 columns]

## **Otázka č. 1 - Koľko je mužov, žien a ostatných používateľov?**

In [ ]:
df1["pohlavie"].value_counts()
# pre túto otázku použijeme jednoduchú funkciu value_counts()

pohlavie
Muž                    29409
Žena                   29157
Nebinárny               1819
Preferuje_nepovedať      608
Name: count, dtype: int64

V datasete je približne rovnaký počet mužov a žien.
Menšiu časť tvoria nebinárni používatelia a používatelia,
ktorí sa rozhodli neuviesť svoje pohlavie.

# **Otázka č.2 - Aký je priemerný vek mužov a žien?**

In [ ]:
df1.groupby("pohlavie")["vek"].mean()


pohlavie
Muž                    39.024100
Nebinárny              38.075972
Preferuje_nepovedať    38.010811
Žena                   38.927139
Name: vek, dtype: float64

Priemerný vek mužov je približne 39.0 roka a priemerný vek žien
približne 38.9 roka. Rozdiel medzi mužmi a ženami je minimálny,
čo naznačuje podobné vekové rozloženie týchto skupín v datasete.

# **Otázka č.3 - V ktorých krajinách ľudia trávia na Instagrame najviac času za deň?**

In [ ]:
df1.groupby("krajina")["priemerna_dlzka_relacie_min"].mean().sort_values(ascending=False)

krajina
Iné                       20.781389
Brazília                  20.635791
Australia                 20.627112
India                     20.541689
Spojené státy americké    20.416094
Južná Koreja              20.374341
Kanada                    20.348265
Nemecko                   20.338924
Spojené královstvo        20.269857
Japonsko                  20.245379
Name: priemerna_dlzka_relacie_min, dtype: float64

Výsledky ukazujú, že najvyšší priemerný čas strávený na Instagrame
má skupina označená ako „Iné“, ktorá však zahŕňa viacero rôznych krajín
a preto nie je možné ju interpretovať ako jednu konkrétnu krajinu.

Spomedzi konkrétnych krajín majú najvyšší priemerný čas používatelia
z Brazílie, Austrálie a Indie. Rozdiely medzi jednotlivými krajinami
sú však relatívne malé, čo naznačuje, že krajina má len obmedzený
vplyv na čas strávený na platforme.

# **Otázka č.4 - Trávia ľudia s vyšším príjmom viac alebo menej času na Instagrame?**

In [ ]:
df1.groupby("uroven_prijmu")["priemerna_dlzka_relacie_min"].mean().sort_values(ascending=False)

uroven_prijmu
Upper-middle    20.801495
High            20.616946
Low             20.437862
Middle          20.416346
Lower-middle    20.364593
Name: priemerna_dlzka_relacie_min, dtype: float64

Výsledky neukazujú výrazný rozdiel v priemernom čase strávenom na Instagrame
medzi jednotlivými príjmovými skupinami. Hoci skupiny s vyšším a upper-middle
príjmom majú mierne vyšší priemer, rozdiely sú minimálne a neindukujú
silnú závislosť medzi príjmom a časom stráveným na platforme.

# **Otázka č.5 - Sú ľudia z miest na Instagrame aktívnejší ako ľudia z vidieka?**

In [ ]:
df1.groupby("mesto_vidiek")["skore_angazovanosti"].mean()

mesto_vidiek
Mesot         1.653844
Predmestie    1.637072
Vidiek        1.612092
Name: skore_angazovanosti, dtype: float64

Výsledky ukazujú, že používatelia z miest majú v priemere mierne vyššie skóre angažovanosti
ako používatelia z vidieka. Rozdiely však nie sú výrazné, čo naznačuje, že typ prostredia
má len obmedzený vplyv na angažovanosť na Instagrame.

# **Otázka č.6 - Ktoré pracovné skupiny (študenti, zamestnaní, nezamestnaní) sú na Instagrame najaktívnejšie?**

In [ ]:
df1.groupby("pracovny_status")["skore_angazovanosti"].mean()

pracovny_status
 Dôchodca        1.622101
Freelancer       1.621880
Full-time        1.657623
Nezamestnanec    1.607561
Part-time        1.626477
Student          1.648021
Name: skore_angazovanosti, dtype: float64

Najvyššie priemerné skóre angažovanosti majú používatelia
pracujúci na plný úväzok a študenti. Rozdiely medzi jednotlivými
pracovnými skupinami sú však relatívne malé, čo naznačuje,
že pracovný status má len obmedzený vplyv na angažovanosť
na Instagrame.

# **Záver**📊

V tomto projekte boli dáta najprv extrahované a vyčistené pomocou Pythonu
a knižnice Pandas v prostredí Google Colab. Na vyčistenom datasete bola
vykonaná základná exploratívna analýza (EDA) a odpovede na analytické otázky.

Vyčistený dataset bol následne uložený do CSV súboru a importovaný
do databázy PostgreSQL. Ďalšie analytické otázky a doplnkové analýzy
boli realizované pomocou SQL v nástroji pgAdmin.

SQL dotazy použité v projekte sú dostupné v samostatných `.sql` súboroch
v tomto repozitári na GitHube.